File: bionic-js-ios/BjsTests/WrappedObjects/ToyWrapped1.swift
import Bjs

class ToyWrapped1: BjsExport {
    
    static var deallocCounter: Int = 0
    var number1String, number2String: String?
    
    class var pi: Double? {
        get { return Double.pi }
    }
    
    class func sum(_ number1: Int?, _ number2: Int?) -> Int? {
        return number1! + number2!
    }
    
    init(_ number1: String?, _ number2: String?) {
        self.number1String = number1
        self.number2String = number2
    }
    
    var number1: Int? {
        get { return Int(number1String!) }
        set { number1String = String(newValue!) }
    }
    
    var number2: Int? {
        get { return Int(number2String!) }
        set { number2String = String(newValue!) }
    }
    
    func getSum(_ offset: Int?) -> Int? {
        return offset! + Int(number1String!)! + Int(number2String!)!
    }
    
    func getToySum(_ toy: ToyWrapped1?) -> Int? {
        return getSum(0)! + toy!.getSum(0)!
    }
    
    deinit {
        ToyWrapped1.deallocCounter += 1
        print("**** dealloc \(ToyWrapped1.deallocCounter)")
    }
}

File: bionic-js-ios/BjsTests/Bjs/test.bundle/src/ToyWrapped1.js
const native = require('ToyWrapped1')

class ToyWrapped1 {
    
    // @bionic Float
    static get pi() {
        return native.bjsStaticGet_pi()
    }
    
    // @bionic (Int, Int) => Int
    static sum(number1, number2) {
        return native.bjsStatic_sum(number1, number2)
    }
    
    // @bionic (String, String)
    constructor(number1, number2) {
        native.bjsBind(this, number1, number2)
    }
    
    // @bionic Int
    get number1() {
        return native.bjsGet_number1(this)
    }
    
    // @bionic Int
    set number1(newValue) {
        native.bjsSet_number1(this, newValue)
    }
    
    // @bionic Int
    get number2() {
        return native.bjsGet_number2(this)
    }
    
    // @bionic Int
    set number2(newValue) {
        native.bjsSet_number2(this, newValue)
    }
    
    // @bionic (Int) => Int
    getSum(offset) {
        return native.bjs_getSum(this, offset)
    }
    
    // @bionic (ToyWrapped1) => Int
    getToySum(toy) {
        return native.bjs_getToySum(this, toy)
    }
}

module.exports = ToyWrapped1
